services/app-api/utils/dynamo/dynamodb-lib.ts

```diff
@@ -1,57 +1,62 @@
 import {
-  DynamoWrite,
-  DynamoDelete,
-  DynamoUpdate,
-  DynamoGet,
-  DynamoScan,
-} from "../types/other";
+  DeleteCommand,
+  DeleteCommandInput,
+  DynamoDBDocumentClient,
+  GetCommand,
+  GetCommandInput,
+  QueryCommand,
+  QueryCommandInput,
+  PutCommand,
+  PutCommandInput,
+} from "@aws-sdk/lib-dynamodb";
+// utils
+import { logger } from "../debugging/debug-lib";
+// types
+import { AnyObject } from "../types";
 
-export function createDbClient() {
-  const dynamoConfig: DynamoDB.DocumentClient.DocumentClientOptions &
-    ServiceConfigurationOptions &
-    DynamoDB.ClientApiVersions = {};
+const localConfig = {
+  endpoint: process.env.DYNAMODB_URL,
+  region: "localhost",
+  credentials: {
+    accessKeyId: "LOCALFAKEKEY", // pragma: allowlist secret
+    secretAccessKey: "LOCALFAKESECRET", // pragma: allowlist secret
+  },
+  logger,
+};
+
+const awsConfig = {
+  region: "us-east-1",
+  logger,
+};
 
-  const endpoint = process.env.DYNAMODB_URL;
-  if (endpoint) {
-    dynamoConfig.endpoint = endpoint;
-    dynamoConfig.credentials = new Credentials({
-      accessKeyId: "LOCALFAKEKEY", // pragma: allowlist secret
-      secretAccessKey: "LOCALFAKESECRET", // pragma: allowlist secret
-    });
-  } else {
-    dynamoConfig["region"] = "us-east-1";
-  }
+export const getConfig = () => {
+  return process.env.DYNAMODB_URL ? localConfig : awsConfig;
+};
 
-  return new DynamoDB.DocumentClient(dynamoConfig);
-}
+const client = DynamoDBDocumentClient.from(new DynamoDBClient(getConfig()));
 
 export default {
-  get: async <Result>(params: DynamoGet) => {
-    const result = await createDbClient().get(params).promise();
-    return { ...result, Item: result?.Item as Result | undefined };
-  },
-  query: (params: any) => createDbClient().query(params).promise(),
-  scan: async <Result>(params: DynamoScan) => {
-    const result = await createDbClient().scan(params).promise();
-    return { ...result, Items: result?.Items as Result[] | undefined };
+  get: async (params: GetCommandInput) => {
+    return await client.send(new GetCommand(params));
   },
-  /**
-   * Scan operation that continues for all results. More expensive but avoids stopping early when a index is not known.
-   */
-  scanAll: async <Result = any>(params: DynamoScan) => {
-    const items = [];
-    let complete = false;
-    while (!complete) {
-      const result = await createDbClient().scan(params).promise();
-      items.push(...((result?.Items as Result[]) ?? []));
-      params.ExclusiveStartKey = result.LastEvaluatedKey;
-      complete = result.LastEvaluatedKey === undefined;
-    }
-    return { Items: items, Count: items.length };
+  delete: async (params: DeleteCommandInput) =>
+    await client.send(new DeleteCommand(params)),
+  put: async (params: PutCommandInput) =>
+    await client.send(new PutCommand(params)),
+  query: async (params: QueryCommandInput) =>
+    await client.send(new QueryCommand(params)),
```

Comment:
Thought: We don't have any code in the API where we couldn't drop in `queryAll` in place of `query`, and I believe any future call to `query` is more likely to be a latent bug than a deliberate choice to ignore extra result pages.

I would vote to consolidate the two methods: to either remove `query` and replace the two remaining calls (both in formTemplates.ts) with `queryAll`, or to rename `queryAll` to `query` (making it the default behavior).